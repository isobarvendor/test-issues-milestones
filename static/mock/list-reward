[ {       "public_code": "shopee-test-2",
       "secret_code": "005ZWWZXFE82",
       "url": "https://barcode-stage.web.app/c/shopee-test-2",
        "secret_code_image": "/img/landing/barcode.png",
       "store_name": "Shopee",
       "coupon_expired_at": "2021-05-09 17:50:00",
       "creative_image": "https://cdn2.grivy.com/Testing%20Juice/JUICE_BANNER-1080x480.jpg"
   },
   {       "public_code": "shopee-test-2",
       "secret_code": "005ZWWZXFE82",
       "url": "https://barcode-stage.web.app/c/shopee-test-2",
        "secret_code_image": "/img/landing/barcode.png",
       "store_name": "Shopee",
       "coupon_expired_at": "2021-05-09 17:50:00",
       "creative_image": "https://cdn2.grivy.com/Testing%20Juice/JUICE_BANNER-1080x480.jpg"
   },
   {       "public_code": "shopee-test-2",
       "secret_code": "005ZWWZXFE82",
       "url": "https://barcode-stage.web.app/c/shopee-test-2",
        "secret_code_image": "/img/landing/barcode.png",
       "store_name": "Shopee",
       "coupon_expired_at": "2021-05-09 17:50:00",
       "creative_image": "https://cdn2.grivy.com/Testing%20Juice/JUICE_BANNER-1080x480.jpg"
   }
]
