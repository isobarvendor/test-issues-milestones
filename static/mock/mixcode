{
                      "programId": 453269,
                      "code": "3WTD90GE5",
                      "lot": {
                        "lotId": 4455843436,
                        "lotName": "Coke 500ml PET",
                        "activateDate": 1611208800000,
                        "inactivateDate": 1627603200000,
                        "active": true,
                        "expired": false,
                        "bevProdPkg": {
                          "trademarkCd": "05",
                          "trademarkName": "Coca-Cola",
                          "brandCd": "001",
                          "brandName": "Coca-Cola",
                          "bevProdCd": "0001",
                          "bevProdName": "Coca-Cola",
                          "caffeinated": true,
                          "carbonated": true,
                          "calorieCategory": {
                            "code": 1,
                            "description": "Regular"
                          }
                        },
                        "releaseId": 494494,
                        "releaseName": "Coke PET",
                        "organizationId": 157168,
                        "organizationName": "Harish",
                        "realtimeCode": false,
                        "testRelease": false,
                        "pointValue": 0,
                        "bevProdOptions": []
                      },
                      "redeemed": false,
                      "actualCode": "3WTD90GE5"
                    }
