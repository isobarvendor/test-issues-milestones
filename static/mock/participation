{"id":16,"questionId":1,"answerId":3,"participationId":17}
