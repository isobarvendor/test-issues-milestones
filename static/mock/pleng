{
  "burnResult": [
    {
      "pincode": "0J6TC2VP5",
      "programId": "453269",
      "lotId": "445580",
      "burned": true
    }
  ],
  "participationId": "1ylazdy9kkxmpa53",
  "instantWinResult": {
    "winner": true,
    "redeemedPrize": {
      "barcodeType": 1,
      "deliveryType": 1,
      "active": true,
      "description": "<p>3 months subscription to Pleng Pro</p>",
      "shortDescription": "<p>ឈ្នះថ្លៃសេវា Pleng Pro Subscription រយៈពេល 3 ខែ</p>",
      "priority": 1,
      "prizeId": "ks8o4ixh",
      "imgUrl": "https://d1tnwdlrciourl.cloudfront.net/prizeImages/z2xtqi9ks8ndwx3/ks8o4ixh/CAM prizes — Pleng Pro@2x.png",
      "name": "3 months subscription to Pleng Pro ",
      "redemptionLink": "https://pleng.onelink.me/pp89?pid=ddlink&af_web_dp=https%3A%2F%2Fwww.smart.com.kh%2Fkm%2Fpleng%2F&af_dp=plengmusic%3A%2F% 2F&af_partner_name=cc",
      "configurationId": "z2xtqi9ks8ndwx3",
      "redeemDescription": "<p>អបអរសាទរ! អ្នកបានឈ្នះ Pleng Pro Subscription រយៈពេល ៣ ខែ! រីករាយនូវការស្តាប់ចម្រៀងឥតដែនកំណត់។ សូមដោនឡូតកម្មវិធី Pleng ពី Smart ដើម្បីបញ្ចូលលេខកូដ និងដាក់ដំណើរការ Smart Pleng Pro។ សូមប្តូរយករង្វាន់ឲ្យបានមុនថ្ងៃទី ៣០ ខែ វិច្ឆិកា ឆ្នាំ ២០២១។</p>",
      "voucherCode": "CNXFUUAKAY"
      
    }
  }
}