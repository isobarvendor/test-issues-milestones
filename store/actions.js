
import GeneralAPI from '../api/general';

import {LOGIN, SIGNUP, GET_ACCOUNT} from './action_types';

export default {
  [LOGIN]: ({ commit, state, getters }, data) => {
    return new Promise((resolve, reject) => {
      const moduleState = state;
      if (moduleState.login) {
        return resolve(moduleState.login)
      } else {
        GeneralAPI.login(data)
        .then(response => {
          commit('SET_TOKEN', response.data.token);
          sessionStorage.token=response.data.token;
          return resolve(messages);
        })
        .catch(error => {

          return reject(error);
        });
      }
    });
  },
  [GET_ACCOUNT]: ({ commit, state, getters }, token) => {
    return new Promise((resolve, reject) => {
      const moduleState = state;
      if (moduleState.login) {
        return resolve(moduleState.login)
      } else {
        GeneralAPI.getAccount(token)
        .then(response => {

          commit('SET_LOGIN_ACCOUNT', response.data);
          return resolve(messages);
        })
        .catch(error => {

          return reject(error);
        });
      }
    });
  },
  [SIGNUP]: ({ commit, state, getters }, data) => {
    return new Promise((resolve, reject) => {
      const moduleState = state;
      if (moduleState.login) {
        return resolve(moduleState.login)
      } else {
        GeneralAPI.signup(data)
        .then(messages => {
          //commit('SET_LOGIN_ACCOUNT', messages);
          return resolve(messages);
        })
        .catch(error => {
          console.error(error);
          return reject(error);
        });
      }
    });
  },


  /*[SET_GLOBAL_ERROR_DIALOG]:({commit, state}, status)=>{
    commit('SET_GLOBAL_ERROR_DIALOG', status);
  }
  */
};
