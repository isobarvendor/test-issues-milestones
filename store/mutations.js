

//shared store are concatenated to the central store
//TO-DO: manage the shared store seperately by module
export default {

  SET_LOGIN_ACCOUNT:(state, data)=>{
        state.login = data;
    },

  SET_TOKEN:(state, data)=>{
      state.token = data;
  },
  SET_ERROR:(state, data)=>{
    state.errorLogin = data;
},
  SET_LIST_PRIZE:(state, data)=>{
    state.listPrize = data;
  },
  SET_LIST_WALLET:(state, data)=>{
    state.listWallet = data;
  },
  SET_FILE_AMAZON:(state, data)=>{
    state.fileUploaded = data;
  },
  SET_CONFIG:(state,data)=>{
    state.config = data
  },
  SET_CMS_CONTENT:(state,data)=>{
    state.CMSContent = data
  }



};
