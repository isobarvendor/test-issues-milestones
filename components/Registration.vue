<template>
<div class="registration">
    <h3>{{data.title}}</h3>
     <div class="registration-body">
    <div class="details">
        <div class="input-container">
            <v-text-field label="Name" name="name" v-model="user.name"  v-validate="'required'">   </v-text-field>
            <span class="error-message">{{ errors.first('name') }}</span>
            <v-text-field label="Email address" name="email"  v-model="user.email"  v-validate="'required|email'" >   </v-text-field>
            <span class="error-message">{{ errors.first('email') }}</span>
            <v-text-field label="Password" name="password"  :type="'password'" v-model="user.password"  v-validate="'required'" >   </v-text-field>
            <span class="error-message">{{ errors.first('password') }}</span>
            <v-text-field label="Confirm password" name="confirm password"  :type="'password'" v-model="cpassword"  v-validate="'required'" >   </v-text-field>
            <span class="error-message">{{ errors.first('confirm password') }}</span>
        </div>
        <div class="radio-container">
            <!-- <v-radio-group v-model="radioGroup">
                <v-radio></v-radio>
            </v-radio-group> -->
            <v-checkbox v-model="terms"></v-checkbox>
            <div class="radio-label"><span v-html="data.privacyRegister.description"></span></div>
        </div>
        <div class="button-container">
          <div class="error-message" v-if="errorMessage" v-html="errorMessage"></div>
          <div class="btn-area">
               <v-progress-circular
                  :width="2"
                  color="white"
                  indeterminate
                  v-if="loading"
                ></v-progress-circular>
            <a  class="button" v-on:click="register()" v-else >Register</a>
          </div>
        </div>

    </div>
    <div class="profile-icon">
            <img src="/img/icons/profile-icon.png"  v-if="!image"/>
      <div v-if="!image" class="input-area">

       <input type="file" @change="onFileChange" value="uploadReceipt">
      </div>
      <div v-else-if="image" class="input-area">
        <img :src="image" width="100%" /><BR/>
        <button @click="removeImage">Remove image</button>
      </div>
    </div>
    </div>
</div>
</template>

<script>
import { SIGNUP, LOGIN, UPLOAD_FILE } from '@/store/action_types';
export default {
  name: "Registration",
  inject: ['$validator'],
  components: {
  },
  data() {
    return {
      terms:false,
      cpassword:null,
      user:{
        name:null,
        email:null,
        password:null,
      },
      errorMessage:null,
      image:null,
      uploadFile:null,
      amazonImage:null,
      loading:false
    };
  },
  props: {
    data: null,
  },
  computed:{
    loginInfo(){
      return this.$store.getters.getLoginAccount;
    }
  },
  methods:{
    onFileChange(e) {
      let files = e.target.files || e.dataTransfer.files;
      if (!files.length)
        return;
      let FileSize = files[0].size / 1024 / 1024; // in MB
        if (FileSize > 2) {
           this.errorMessage ="Please upload file not more than 2 MB"
           return;
        }
      let allowedExtensions = /(\.jpg|\.jpeg|\.png|\.gif)$/i;
      let filePath = e.target.value;
      if(!allowedExtensions.exec(filePath)){
         this.errorMessage ="Please upload image file"
           return;
      }
      this.uploadFile = files[0];
      this.createImage(files[0]);
       this.errorMessage = null;
    },
    createImage(file) {
      var image = new Image();
      var reader = new FileReader();
      var vm = this;

      reader.onload = (e) => {
        vm.image = e.target.result;
      };
      reader.readAsDataURL(file);
    },
    removeImage: function (e) {
      this.image = '';

    },
      async register(){

           this.$validator.validateAll().then(async(valid) => {

         if(this.loginInfo){
            localStorage.clear();
          }
         if(valid){
            this.loading=true;
             this.errorMessage=null;
             if(!this.terms){
               this.loading=false;
              this.errorMessage="Please accept the terms and condition"
              return false;
            }
            if(this.user.password!=this.cpassword){
              this.loading=false;
              this.errorMessage="Your password and confirm password is different"
              return false;
            }
              if(this.uploadFile){
              var formData = new FormData();
              formData.append("file", this.uploadFile);
                let upload={
                request:formData,
                type:'profile-picture'
              }

               await this.$store.dispatch(UPLOAD_FILE,upload)
               .then((response)=>{

                  this.amazonImage=response.data.url;

                })
                .catch((error) =>{
                });
            }
             if(this.amazonImage){
                 this.user.profilePicture=this.amazonImage;
               }

            this.$store.dispatch(SIGNUP,this.user)
            .then( (response)=> {
              let login = {
                email:this.user.email,
                password:this.user.password
              }

               this.$store.dispatch(LOGIN,login)
                .then((response)=>{
                    this.loading=false;
                    location.href='/account';
                })
                 .catch((error) =>{
                  this.loading=false;
                  if(error.response){
                    this.errorMessage='Something went wrong please try again';
                  }
            })
            })
            .catch((error) =>{
              this.loading=false;
              if(error.response && error.response.data.status=='400'&&error.response.data.message=="error.validation"){
                this.errorMessage='Please use another email';
              }else{
                this.errorMessage='Something went wrong please try again';
              }
            })
         }

       });
      },
      radioGroup(){

      }
  },
  beforeMount() {},
  mounted(){

  }
};
</script>

<style lang="scss">
.registration{

  .profile-icon{
    position: relative;
     input{
      width: 250px;
      height: 250px;
      position: absolute;
      opacity: 0;
      top:0px;

    }
    .input-area{
      text-align: center;
        img{
                width:235px;
                height: 235px;
                border-radius: 120px;
            }
    }

  }

    .error-message{
      color:red;
    }
    h3{
        font-size: 25px;
        text-transform: none;
    }
    .registration-body{
        display: flex;
        justify-content: space-between;
        align-items: center;
        width: 85%;
        .v-text-field__slot{
            border-bottom: 1px solid #585858;
        }
        .theme--light.v-label{
            color: #585858;
        }
        .theme--light.v-input input{
            color: #ffffff;
        }
        .details{
            margin-top: 40px;
            width: 50%;
        }
        .input-container{
            margin-top: 25px;
        }
        .radio-container{
            display: flex;
            align-items: center;
            .radio-label{
                font-size: 14px;
                margin-left: 10px;
                line-height: 24px;
            }
            .v-input--selection-controls__input .v-icon{
                color: #585858;
            }
        }
        .button-container{
            .btn-area{
              text-align: center;
            }
            margin-top: 40px;
            a.button{
                width: 230px;
                display: block;
            }
        }

    }
    @media only screen and (max-width: 1199px) {
        .registration-body{
            flex-direction: column-reverse;
            width: 100%;
            .details{
                width: 100%;
            }
            .profile-icon{
                margin-top: 35px;
            }
            a.button{
                margin:auto;
            }
        }
        h3{
            text-align: center;
            font-size: 22px;
        }
    }
}
</style>
