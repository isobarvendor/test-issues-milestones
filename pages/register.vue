<template>
  <div v-if="data" id="main" class="register">
    <div class="wrapper">
      <div class="container">
        <Registration :data="data[0].signUp"/>
      </div>

    </div>
  </div>
</template>

<script>
import Registration from '@/components/Registration'
export default {
  data(){
    return{

    }
  },
  head() {
    return {
      title: "Coke Campaign Title",
      meta: [
        {
          hid: "description",
          name: "description",
          content: "Home page description"
        }
      ],
      noscript: [],
     /*script: [
        {
          hid: "anime",
          src: "/js/anime.min.js"
        }
      ],*/
      css: []
    };
  },
  computed: {
    data(){
        return this.$store.getters.getCMSContent;
     }
    },
};
</script>

<style lang="scss">
@import '~assets/scss/variables.scss';
  html, body{
    background: $body-bg-color-wallet;
    @media only screen and (max-width: 767px) {
      .register{
      }
    }
  }
</style>
