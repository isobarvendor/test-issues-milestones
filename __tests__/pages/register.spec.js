import { mount,  createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import VueRouter from 'vue-router'

import register from '@/pages/register';

import Registration from '@/components/Registration'


import {mockCMSConfig,mockCMSContent} from "@/__mock__/mock";


import actions from '@/store/actions';
import mutations from '@/store/mutations';


let localVue, store,router;


  beforeEach(() => {
    localVue = createLocalVue();
    localVue.use(Vuex);
    localVue.use(VueRouter);
    router = new VueRouter();
    store = new Vuex.Store({
      getters: {
        getCMSContent() {
          return mockCMSContent;
        },
        getLoginAccount(){
          return {
            name:"ron",
            email:"ron@dsw.com",
            imageUrl:"tes"
          }
        },
        getCMSConfig(){
          return mockCMSConfig[0];
        }
      },
      state:{
        token:"",
        language: "en",
        listWallet:{
          walletStatus:[
            {amount :30}
          ]
        },

      },
      actions,
      mutations
    });
  });

  const factory = () => {
    return mount(register, {

      localVue, store,router
    });
  };



describe('register', () => {


  test('renders a Registration', () => {
    const wrapper = factory();
    const component= wrapper.findComponent(Registration);
    expect(component.exists()).toBeTruthy();
  });


  test("mounts properly", () => {
    const wrapper = factory();
    expect(wrapper.vm).toBeTruthy();
  });

  test("renders properly", () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  });

});
