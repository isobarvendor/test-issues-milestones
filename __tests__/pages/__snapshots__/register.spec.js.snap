// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`register renders properly 1`] = `
<div id="main" class="register">
  <div class="wrapper">
    <div class="container">
      <div class="registration">
        <h3>Register for an account</h3>
        <div class="registration-body">
          <div class="details">
            <div class="input-container">
              <div class="v-input theme--light v-text-field">
                <div class="v-input__control">
                  <div class="v-input__slot">
                    <div class="v-text-field__slot"><label for="input-48" class="v-label theme--light" style="right: 0px; position: absolute;">Name</label><input name="name" id="input-48" type="text"></div>
                  </div>
                  <div class="v-text-field__details">
                    <div class="v-messages theme--light">
                      <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                    </div>
                  </div>
                </div>
              </div> <span class="error-message"></span>
              <div class="v-input theme--light v-text-field">
                <div class="v-input__control">
                  <div class="v-input__slot">
                    <div class="v-text-field__slot"><label for="input-51" class="v-label theme--light" style="right: 0px; position: absolute;">Email address</label><input name="email" id="input-51" type="text"></div>
                  </div>
                  <div class="v-text-field__details">
                    <div class="v-messages theme--light">
                      <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                    </div>
                  </div>
                </div>
              </div> <span class="error-message"></span>
              <div class="v-input theme--light v-text-field">
                <div class="v-input__control">
                  <div class="v-input__slot">
                    <div class="v-text-field__slot"><label for="input-54" class="v-label theme--light" style="right: 0px; position: absolute;">Password</label><input name="password" id="input-54" type="password"></div>
                  </div>
                  <div class="v-text-field__details">
                    <div class="v-messages theme--light">
                      <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                    </div>
                  </div>
                </div>
              </div> <span class="error-message"></span>
              <div class="v-input theme--light v-text-field">
                <div class="v-input__control">
                  <div class="v-input__slot">
                    <div class="v-text-field__slot"><label for="input-57" class="v-label theme--light" style="right: 0px; position: absolute;">Confirm password</label><input name="confirm password" id="input-57" type="password"></div>
                  </div>
                  <div class="v-text-field__details">
                    <div class="v-messages theme--light">
                      <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                    </div>
                  </div>
                </div>
              </div> <span class="error-message"></span>
            </div>
            <div class="radio-container">
              <div class="v-input theme--light v-input--selection-controls v-input--checkbox">
                <div class="v-input__control">
                  <div class="v-input__slot">
                    <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate material-icons theme--light">$checkboxOff</i><input aria-checked="false" id="input-60" role="checkbox" type="checkbox" value="">
                      <div class="v-input--selection-controls__ripple"></div>
                    </div>
                  </div>
                  <div class="v-messages theme--light">
                    <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                  </div>
                </div>
              </div>
              <div class="radio-label"><span>- - I accept the <a href="/tnc">Terms and Conditions</a> &amp; <a href="/privacy">Privacy Policy</a> for sign up.</span></div>
            </div>
            <div class="button-container">
              <!---->
              <div class="btn-area"><a class="button">Register</a></div>
            </div>
          </div>
          <div class="profile-icon"><img src="/img/icons/profile-icon.png">
            <div class="input-area"><input type="file" value="uploadReceipt"></div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
